### DISCUSSION TITLE
clip_box behavior

### DISCUSSION THREAD

**Discussion by eino** [pyvista/pyvista#2191] [Q&A] [answered]
Hi all,
can anyone help me understand the behavior of `clip_box`? The result of clipping a mesh with a Cube object doesn't seem to make sense to me.
For instance the behavior of clipping the plane with the red cube, from the doc itself: https://docs.pyvista.org/_images/sphx_glr_clipping_004.png (from [here](https://docs.pyvista.org/examples/01-filter/clipping.html#clip-with-plane-box-example))
Why is there a part of the plane on the left not clipped? Why is there a part inside the cube clipped?
It's the same with other tries I've made.

Can anyone explain the logic? Or is `clip_box` broken?
Thanks in advance

**Comment by banesullivan** [comment, marked as answer]
> Why is there a part of the plane on the left not clipped? Why is there a part inside the cube clipped?
It's the same with other tries I've made.

😨 Agh!! That is NOT supposed to be like that. That's a serious bug/regression in either PyVista or VTK that we need to look into.

**Reply by banesullivan** [reply]
@pyvista/developers, we may want to go back to regression testing with screenshots. I will look into this and see if it is a PyVista or VTK regression

**Reply by banesullivan** [reply]
I can confirm that this is a PyVista regression across versions of VTK which was introduced in 0.33.0. I will do a git bisect to find the offending commit. In the meantime, I suspect it has something to do with the changes to the `inplace` argument for this particular example, but I will continue to investigate.

**Reply by adeak** [reply]
@banesullivan there's no "go back to" regression testing with screenshots: we do that. For testing, not examples.

**Reply by eino** [reply]
@banesullivan thank you for looking into it. If it can be of any help, here's a few pieces of info I gathered during my investigations. From what I saw it doesn't seem to be linked to the `inplace` argument, as clipping is broken even if the whole `rotate_z` line is removed:
![Screenshot from 2022-02-14 09-10-31](https://user-images.githubusercontent.com/4985/153824908-8a70af99-54a4-4faa-b794-c2175d93f1c3.png)


Also, I suspected the `Cube` object, but clipping is also broken (albeit differently) if we pass `roi.bounds` to the clipping method:
![Screenshot from 2022-02-14 09-11-48](https://user-images.githubusercontent.com/4985/153825022-6e95c669-6dd4-49fd-b45a-7bfc9ccdbd92.png)

On the other way, if we both don't rotate, and pass `roi.bounds`, then clipping looks sane. Not sure if anything can be deduced from it?
![Screenshot from 2022-02-14 09-14-16](https://user-images.githubusercontent.com/4985/153825474-277985ba-4a1e-4514-a78b-4bb03821bb56.png)

**Reply by banesullivan** [reply]
Following up that this was solved in #2314

**Comment by RichardScottOZ** [comment]
Is it just a clip box problem?
